File: pkg/server/api/mcp/v1/types/people.go
// Copyright (c) 2025 Probo Inc <hello@getprobo.com>.
//
// Permission to use, copy, modify, and/or distribute this software for any
// purpose with or without fee is hereby granted, provided that the above
// copyright notice and this permission notice appear in all copies.
//
// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
// REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
// AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
// INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
// LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
// OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR

package types

import (
	"go.probo.inc/probo/pkg/coredata"
	"go.probo.inc/probo/pkg/mail"
	"go.probo.inc/probo/pkg/page"
)

func NewPeople(p *coredata.People) *People {
	additionalEmails := p.AdditionalEmailAddresses
	if additionalEmails == nil {
		additionalEmails = []mail.Addr{}
	}
	return &People{
		FullName:                 p.FullName,
		ID:                       p.ID,
		OrganizationID:           p.OrganizationID,
		Kind:                     p.Kind,
		PrimaryEmailAddress:      p.PrimaryEmailAddress,
		AdditionalEmailAddresses: additionalEmails,
		Position:                 p.Position,
		ContractStartDate:        p.ContractStartDate,
		ContractEndDate:          p.ContractEndDate,
		CreatedAt:                p.CreatedAt,
		UpdatedAt:                p.UpdatedAt,
	}
}

func NewListPeopleOutput(peoplePage *page.Page[*coredata.People, coredata.PeopleOrderField]) ListPeopleOutput {
	people := make([]*People, 0, len(peoplePage.Data))
	for _, p := range peoplePage.Data {
		people = append(people, NewPeople(p))
	}

	var nextCursor *page.CursorKey
	if len(peoplePage.Data) > 0 {
		cursorKey := peoplePage.Data[len(peoplePage.Data)-1].CursorKey(peoplePage.Cursor.OrderBy.Field)
		nextCursor = &cursorKey
	}

	return ListPeopleOutput{
		NextCursor: nextCursor,
		People:     people,
	}
}

File: packages/helpers/src/people.ts
// Copyright (c) 2025-2026 Probo Inc <hello@probo.com>.
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
// SOFTWARE.

type Translator = (s: string) => string;

export const peopleRoles = [
    "EMPLOYEE",
    "CONTRACTOR",
    "SERVICE_ACCOUNT",
] as const;

export function getRoles(t: Translator) {
    return [
        {
            value: "EMPLOYEE",
            label: t("helpers.peopleRole.employee"),
        },
        {
            value: "CONTRACTOR",
            label: t("helpers.peopleRole.contractor"),
        },
        {
            value: "SERVICE_ACCOUNT",
            label: t("helpers.peopleRole.serviceAccount"),
        },
    ];
}

export function getRole(t: Translator, role?: string): string {
    switch (role) {
        case "EMPLOYEE":
            return t("helpers.peopleRole.employee");
        case "CONTRACTOR":
            return t("helpers.peopleRole.contractor");
        case "SERVICE_ACCOUNT":
            return t("helpers.peopleRole.serviceAccount");
        default:
            return t("helpers.common.unknown");
    }
}
